### PR TITLE
feat: endpoint to check server configuration

### LINKED ISSUE #2818: backend: expose important config to frontend
some config needs to be exposed to the user. This includes org-specific settings like "what are the org's password requirements for users?", but also some server-specific settings like "Is email configured?"

Create an unauthenticated endpoint (perhaps `GET /api/configuration`) that exposes non-sensitive information the client will need. In this case, a setting such as `{isEmailConfigured: true}`

### PR DESCRIPTION
**Pull request by ssoroka**
## Summary

expose server configuration to the frontend so it can make intelligent UI choices. 

## Checklist

- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #2818

### CODE REVIEW

**Comment by jmorganca** [comment]
Great stuff. This may become doubly useful for feature-flagging upcoming features.

File: internal/server/configuration.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package server
+
+import (
+	"github.com/gin-gonic/gin"
+
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/server/email"
+)
+
+func (a *API) GetServerConfiguration(c *gin.Context, _ *api.EmptyRequest) (*api.ServerConfiguration, error) {
```
Comment (BruceMacD) [inline internal/server/configuration.go:10]: I feel like this should be part of the `/api/settings` endpoint, its a bit weird to have both `/api/settings` and `api/server-config`. They sound the same to me.

Comment (ssoroka) [reply internal/server/configuration.go:10]: one is server-specific configuration, one is org-specific settings.

Comment (BruceMacD) [reply internal/server/configuration.go:10]: `/settings` is only for password requirements at the moment, which will be org specific also

Comment (ssoroka) [reply internal/server/configuration.go:10]: that's the org-specific one

File: internal/server/testdata/openapi3.json [hunk 2]
```diff
@@ -4489,6 +4499,91 @@
         ]
       }
     },
+    "/api/server-config": {
```
Comment (pdevine) [inline internal/server/testdata/openapi3.json:4502]: Still need to update this route.

**Review by pdevine — APPROVED** [review]
Totally went back and forth on the API route, but I think `/api/server-configuration` probably is the best bet. Looks like the swagger still needs to be updated.

**Pull request merged**